### PR TITLE
Add ellipsis operator scope

### PR DESCRIPTION
**Pull request by adamkvitek**
# Thanks for improving Semgrep Docs 😀

### Please ensure

- [x] A subject matter expert (SME) reviews the content
- [x] A technical writer reviews the content or PR
- [x] This change has no security implications or else you have pinged the security team
- [x] Redirects are added if the PR changes page URLs

### CODE REVIEW

File: docs/writing-rules/pattern-syntax.mdx [hunk 1]
```diff
@@ -118,13 +118,13 @@ def parse_data(parser, data={}):  # Oops, mutable default arguments
 The YAML `|` operator allows for [multiline strings](https://yaml-multiline.info/).
 :::
 
-The ellipsis operator can also be used for the function name. Indeed,
-In some cases, you may want to match any function definitions:
-regular functions, methods, but also anonymous functions (a.k.a. lambdas).
+The ellipsis operator can also be used for the function name.
+In some cases, you can match any function definitions:
```
Comment (IagoAbal) [inline docs/writing-rules/pattern-syntax.mdx:122]: Do you need the "In some cases" ?

Comment (adamkvitek) [reply docs/writing-rules/pattern-syntax.mdx:122]:
```suggestion
Match any function definition:
```

File: docs/writing-rules/pattern-syntax.mdx [hunk 2]
```diff
@@ -156,6 +156,14 @@ class DataRetriever(InsecureBaseClass):
...
 :::
 
+#### Ellipsis operator scope
+
+The `...` ellipsis operator matches everything in its sub-scope. See the following example:
+
+<iframe src="https://semgrep.dev/embed/editor?snippet=zZx0" border="0" frameBorder="0" width="100%" height="432"></iframe>
+
+Semgrep matches the first occurrence of `bar` and `baz` in the test code as these objects fall under the scope of `foo` and `...`. The ellipsis operator does not match the second occurrence of `bar` and `baz` as they are not inside of the function definition, therefore these objects in their second occurrence are not inside the scope of the ellipsis operator.
+
```
[the remark this answers is not in the document]
Comment (adamkvitek) [reply docs/writing-rules/pattern-syntax.mdx:166]: @IagoAbal thanks! What do you think about the question made here: https://github.com/returntocorp/semgrep-docs/pull/721/files#r981396851 Where should we move this information?

To your last comment: It now sounds like a circular definition. Not sure if this is helping much... The information is specified this way below already.

```suggestion
#### Ellipsis operator scope

The `...` ellipsis operator matches everything in the sub-scope of the code that Semgrep matched. See the following example:

<iframe src="https://semgrep.dev/embed/editor?snippet=zZx0" border="0" frameBorder="0" width="100%" height="432"></iframe>

Semgrep matches the first occurrence of `bar` and `baz` in the test code as these objects fall under the scope of `foo` and `...`. The ellipsis operator does not match the second occurrence of `bar` and `baz` as they are not inside of the function definition, therefore these objects in their second occurrence are not inside the scope of the ellipsis operator.

```

[the remark this answers is not in the document]
Comment (IagoAbal) [reply docs/writing-rules/pattern-syntax.mdx:166]: Maybe:

The `...` ellipsis operator matches everything in the current scope (or sub-scopes). The current scope is given by the code matched by the patterns preceding the `...`. See the following example:

**Review by IagoAbal — APPROVED** [review]

File: docs/writing-rules/pattern-syntax.mdx (hunk 2 above)
[the remark this answers is not in the document]
Comment (adamkvitek) [reply docs/writing-rules/pattern-syntax.mdx:166]:
```suggestion
#### Ellipsis operator scope

The `...` ellipsis operator matches everything in its current scope. The current scope of this operator is defined by the patterns that precede `...` in a rule. See the following example:

<iframe src="https://semgrep.dev/embed/editor?snippet=zZx0" border="0" frameBorder="0" width="100%" height="432"></iframe>

Semgrep matches the first occurrence of `bar` and `baz` in the test code as these objects fall under the scope of `foo` and `...`. The ellipsis operator does not match the second occurrence of `bar` and `baz` as they are not inside of the function definition, therefore these objects in their second occurrence are not inside the scope of the ellipsis operator.
```

**Review by s-santillan — APPROVED** [review]
lgtm! I understood everything. Thank you!

**Pull request merged**